btree/btree.c

```diff
@@ -2039,36 +2091,149 @@ static void generic_move(struct nd *src, struct nd *tgt,
 	node_fix(tgt, tx);
 }
 
+/**
+ * calc_shift is used to calculate the shift for the given number of bytes.
+ * Shift is the exponent of nearest power-of-2 value greater than or equal to
+ * number of bytes. 
+ * 
+ * @param value It represents the number of bytes
+ * @return int returns the shift value.
+ */
+
+int calc_shift(int value)
+{
+	unsigned int sample 	= (unsigned int) value;
+	unsigned int pow 	= 0;
+
+	while (sample > 0)
+	{
+		sample >>=1;
+		pow += 1;
+	}
+
+	return pow - 1;
+}
+
+/**
+ * btree_create_tick function is the main function used to create btree.
+ * It traverses through multiple states to perform its operation.
+ * 
+ * @param smop Represents the state machine operation
+ * @return int64_t It returns the next state to be executed.
+ */
+
+int64_t btree_create_tick(struct m0_sm_op *smop)
+{
+	struct m0_btree_op 	*bop = M0_AMB(bop,smop,bo_op);
+	struct m0_btree_oimpl 	*oi = bop->bo_i;
+	struct td		*tree;
+	struct m0_btree_idata 	*data;
+	struct segaddr 		 curr_addr;
+	int 			 k_size = 8;
+	int 			 v_size = 8;
+
+	switch(bop->bo_op.o_sm.sm_state) 
+	{
+		case P_INIT:
+			data = &bop->b_data;
+
+			bop->bo_i = m0_alloc(sizeof(struct m0_btree_oimpl));
```

Comment:
you cab use sizeof *bop->bo_i instead of sizeof(struct m0_btree_oimpl) as it is recommended in [https://github.com/Seagate/cortx-motr/blob/btree/doc/coding-style.md](https://github.com/Seagate/cortx-motr/blob/btree/doc/coding-style.md)

Reply:
Updated.